apps/src/code-studio/components/libraries/LibraryManagerDialog.jsx

```diff
@@ -222,18 +239,25 @@ export class LibraryManagerDialog extends React.Component {
   };
 
   displayProjectLibraries = () => {
-    const {projectLibraries} = this.state;
+    const {projectLibraries, updatedLibraryChannels} = this.state;
     if (!Array.isArray(projectLibraries) || !projectLibraries.length) {
       return <div style={styles.message}>{i18n.noLibrariesInProject()}</div>;
     }
+
     return projectLibraries.map(library => {
+      // Only pass onUpdate prop for libraries with updates available.
+      let onUpdate;
+      if (updatedLibraryChannels.includes(library.channelId)) {
+        onUpdate = channelId => {
```

Comment:
This callback function is the same for every library, right? I'm wondering if it would be more clear to just declare the function once, outside the map(), then in the map you could use a ternary operator to only pass it as a prop if the library has been updated?
Something like:
```
const onUpdate = channelId => {...}
return projectLibraries.map(library => 
  <LibraryListItem
    onUpdate={updatedLibraryChannels.includes(library.channelId) ? onUpdate : undefined}
    ... />
)
```

That seems potentially better to me because it's a little more clear that the callback *itself* isn't changing at all, just the decision of whether to pass it in? Idk, let me know what you think!

Reply:
i totally agree! i'll make that update now

Reply:
oh, actually, i have a bunch of changes to this in #34004, so i'll make this change there